src/main/java/com/solvd/webtest/pages/ios/SearchResultPageIos.java

```diff
@@ -0,0 +1,67 @@
+package com.solvd.webtest.pages.ios;
+
+import com.solvd.webtest.components.ResultMobile;
+import com.solvd.webtest.pages.base.SearchResultPageBase;
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.*;
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = SearchResultPageBase.class)
+public class SearchResultPageIos extends SearchResultPageBase {
+    @FindBy(css = ".srp-results > li:not(.srp-river-answer)  .s-item__wrapper")
+    private List<ResultMobile> results;
+
+    @FindBy(css = ".srp-controls__control--link.srp-controls__control--link-right.srp-controls__control--link-enabled.btn")
+    private ExtendedWebElement sortingMenuBtn;
+
+    @FindBy(css = "ul.srp-sort__menu span.s-sort-label")
+    private List<ExtendedWebElement> sortOptions;
+
+    public SearchResultPageIos(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public boolean isOpen() {
+        super.isOpen();
+        return getDriver().getCurrentUrl().startsWith(url);
+    }
+
+    public void printResults() {
+        for (var r : results)
+            LOGGER.info(r.print());
+    }
+
+    public List<ResultMobile> getResults() {
+        return results;
+    }
+
+    public void setSortOption(String option) {
+        sortingMenuBtn.click();
+
+        if (!sortOptions.isEmpty())
+            for (var o : sortOptions)
+                if (o.getText().equals(option)) {
+                    o.click();
+                    results.clear();
+                    for (var e : findExtendedWebElements(By.cssSelector(".srp-results > li:not(.srp-river-answer)  .s-item__wrapper")))
```

Comment:
cssSelector will fail if there is any change in the html structure

Reply:
I don`t think there is better locator